src/processTargets/modifiers/scopeHandlers/TokenScopeHandler.ts

```diff
@@ -0,0 +1,79 @@
+import { Position, Range, TextEditor } from "vscode";
+import { getMatcher } from "../../../core/tokenizer";
+import { Target } from "../../../typings/target.types";
+import { getTokensInRange } from "../../../util/getTokensInRange";
+import { expandToFullLine } from "../../../util/rangeUtils";
+import { TokenTarget } from "../../targets";
+import { ContainingIndices, ScopeHandler } from "./scopeHandler.types";
+
+export default class TokenScopeHandler extends ScopeHandler {
+  protected getEveryTarget(
+    editor: TextEditor,
+    contentRange: Range,
+    isReversed: boolean,
+    hasExplicitRange: boolean
+  ): Target[] {
+    const tokenRanges = getTokensInRange(
+      editor,
+      expandToFullLine(editor, contentRange)
+    ).map(({ range }) => range);
+
+    const filteredTokenRanges = hasExplicitRange
+      ? this.filterRangesByIterationScope(contentRange, tokenRanges)
+      : tokenRanges;
+
```

Comment:
Maybe` but I'm not sure if that is going to be consistent across all different scope types.

Reply:
We might have multiple iteration scopes. Functions inside/outside classes for example. I'm not sure it will differ but it night since we haven't thought about that implementation yet it's hard to say.

Also this line is in the middle of another function I'm unsure about how much code we save by extracting this.

Reply:
Also allows us to make `filterRangesByIterationScope` private rather than protected, which is nice.  Just trying to simplify the contract between base class and scope class as much as possible

Reply:
I know you view it as an implementation detail, but I think it should be a first-class api.  For example, we could imagine third parties might implement their own scope types via extensions.  I know that sounds far out, but the more we can view separate components of cursorless as independent pieces with well-defined api contracts, the better

Reply:
I will give it a try

Reply:
Looking at the code more, I'm actually tempted to move the filter stage into the modifiers.  It's only "every" and ordinal that need to do this; containing and relative don't filter

If you're really concerned about the duplication, we could revert to having ordinal call every stage; I actually don't think that's too crazy

Reply:
Updated now. Definitely needs to have a look at the typing system, but something general is in place now at least.